framework/source/class/qx/event/handler/GestureCore.js

```diff
@@ -682,27 +718,125 @@ qx.Bootstrap.define("qx.event.handler.GestureCore", {
+     * introduces a lot more wheel events fired from the trackpad, so the roll factor to be applied
+     * has to be reduced in order to make the scrolling less sensitive.
+     *
+     * @param domEvent {Event} DOM event
      * @param type {String} The type of the dom event
      * @param target {Element} event target
      */
     _fireRoll : function(domEvent, type, target) {
       if (domEvent.type === qx.core.Environment.get("event.mousewheel").type) {
-        domEvent.delta = {
-          x: qx.util.Wheel.getDelta(domEvent, "x") * qx.event.handler.GestureCore.ROLL_FACTOR,
-          y: qx.util.Wheel.getDelta(domEvent, "y") * qx.event.handler.GestureCore.ROLL_FACTOR
-        };
-        domEvent.delta.axis = Math.abs(domEvent.delta.x / domEvent.delta.y) < 1 ? "y" : "x";
-        domEvent.pointerType = "wheel";
+        if (this._isTouchPadDetectionEnabled()) {
+          var now = new Date().getTime();
+          if (this.__lastRollEventTime > 0 && now - this.__lastRollEventTime
+            > qx.event.handler.GestureCore.TOUCHPAD_WHEEL_EVENTS_TIMEOUT) {
+            // The timeout was reached. A new detection step should occur.
+            this.__touchPadDetectionPerformed = false;
+            this.__rollEvents = [];
+            this.__lastRollEventTime = 0;
+          }
+          if (!this.__touchPadDetectionPerformed) {
+            // We are into a detection session. We count the events so that we can decide if
+            // they were fired by a real mouse wheel or a touchpad. Just swallow them until the
+            // detection period is over.
+            if (this.__rollEvents.length === 0) {
+              // detection starts
+              this.__rollEventsCountStart = now;
+              qx.event.Timer.once(function () {
+                if (!this.__touchPadDetectionPerformed) {
+                  // There were not enough events during the TOUCHPAD_WHEEL_EVENTS_PERIOD to actually
+                  // trigger a scrolling. Trigger it manually.
+                  this.__endTouchPadDetection();
+                  this.__performAdaptativeRollScrolling(target);
+                }
+              }, this, qx.event.handler.GestureCore.TOUCHPAD_WHEEL_EVENTS_PERIOD + 50)
+            }
+            this.__rollEvents.push(domEvent);
+            this.__rollEventsCount++;
+            if (now - this.__rollEventsCountStart > qx.event.handler.GestureCore.TOUCHPAD_WHEEL_EVENTS_PERIOD) {
+              this.__endTouchPadDetection();
+            }
+          }
+          if (this.__touchPadDetectionPerformed) {
+            if (this.__rollEvents.length === 0) {
+              this.__rollEvents.push(domEvent);
+            }
+            // Detection is done. We can now decide the roll factor to apply to the delta.
+            // Default to a real mouse wheel event as opposed to a touchpad one.
+            var event = this.__performAdaptativeRollScrolling(target);
```

Comment:
Travis CI and the linter of the build process complains about `var event` not being used. This should be corrected as else travis will never succeed. See: https://travis-ci.org/qooxdoo/qooxdoo/jobs/214923160 (at the end of the log).